hack/verify-no-vendor-cycles.sh

```diff
@@ -18,14 +18,15 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 
-staging_repos=($(ls "${KUBE_ROOT}/staging/src/k8s.io/"))
+IFS=" " read -r -a staging_repos <<< "$(ls "${KUBE_ROOT}/staging/src/k8s.io/")"
 staging_repos_pattern=$(IFS="|"; echo "${staging_repos[*]}")
 
 failed=false
-for i in $(find vendor/ -type d); do
-  deps=$(go list -f '{{range .Deps}}{{.}}{{"\n"}}{{end}}' ./$i 2> /dev/null || echo "")
+while IFS= read -r -d '' i
```

Comment:
concerned about this change in logic. would love to see this tested to ensure it works as intended.